packages/renderer/src/lib/preferences/PreferencesResourcesRendering.spec.ts

```diff
@@ -140,6 +140,24 @@ test('Expect to see elements regarding foo provider', async () => {
   expect(button).toHaveTextContent('Connect ...');
 });
 
+test('Expect to scroll to the focused element if focus prop is provided', async () => {
+  // Mock the scrollIntoView function
+  const scrollIntoViewMock = vi.fn();
+  window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;
+  const customProviderInfo: ProviderInfo = {
+    ...providerInfo,
+    id: 'test-provider',
+    name: 'Test Provider',
+  };
+  providerInfos.set([customProviderInfo]);
+  render(PreferencesResourcesRendering, { focus: 'test-provider' });
+  await vi.waitFor(() => {
+    // Check if scrollIntoView was called
+    expect(scrollIntoViewMock).toHaveBeenCalledWith({ behavior: 'auto', block: 'start' });
+  });
+  scrollIntoViewMock.mockRestore();
```

Comment:
nit: you can directly set the mock, and use expect on it,

```suggestion
  window.HTMLElement.prototype.scrollIntoView = vi.mock();
  const customProviderInfo: ProviderInfo = {
    ...providerInfo,
    id: 'test-provider',
    name: 'Test Provider',
  };
  providerInfos.set([customProviderInfo]);
  render(PreferencesResourcesRendering, { focus: 'test-provider' });
  await vi.waitFor(() => {
    // Check if scrollIntoView was called
    expect(window.HTMLElement.prototype.scrollIntoView).toHaveBeenCalledWith({ behavior: 'auto', block: 'start' });
  });
```